process_duty_email.py

```diff
@@ -0,0 +1,293 @@
+            os.mkdir(path_to_RPKM)
+            os.mkdir(destination_of_Coverage)
+            # import csv file to pandas df
+            csv_file = pandas.read_csv(file_path, index_col=None)
+            # get data frames from the master df
+            df_RPKM = csv_file[csv_file['type'] == 'RPKM']
+            df_coverage = csv_file[csv_file['type'] == 'coverage']
+            df_FHPRS = csv_file[csv_file['type'] == 'FH_PRS']
+            # Create a folder with NGS run name, subfolders cold coverasge and RPKM, download data
+            results = re.search(r"\\(NGS\S+)$", output_folder)
+            foldername = results.group(1)
+            StG_transfer_folder = StG_transfer + "\\" + foldername
+            # Make directories in the outgoing folder
+            os.mkdir(StG_transfer_folder)
+            os.mkdir(StG_transfer_folder + "\\coverage")
+            os.mkdir(StG_transfer_folder + "\\RPKM")
+            # Download data for StG Transfer
+            url_RPKM = get_data(df_RPKM, '"{}"'.format(path_to_RPKM))
+            url_coverage = get_data_GSTT(df_coverage, '"{}"'.format(destination_of_Coverage))
+            url_StG_RPKM = get_data(df_RPKM, '"{}"'.format(StG_transfer_folder + "\\RPKM"))
+            url_StG_coverage = get_data_StG(df_coverage, '"{}"'.format(StG_transfer_folder + "\\coverage"))
+            if len(df_FHPRS.index) >= 1 :
+                destination_of_FHPRS = output_folder + "\\FH_PRS"
+                os.mkdir(destination_of_FHPRS)
+                os.mkdir(StG_transfer_folder + "\\FH_PRS")
+                url_FHPRS = get_data_GSTT(df_FHPRS, '"{}"'.format(destination_of_FHPRS))
+                url_StG_FHPRS = get_data_StG(df_FHPRS, '"{}"'.format(StG_transfer_folder + "\\FH_PRS"))
+            else:
+                url_FHPRS = ''
+                url_StG_FHPRS = ''
+            all_url =  url_RPKM + url_coverage + url_FHPRS + url_StG_RPKM + url_StG_coverage + url_StG_FHPRS
+            output = download_data(all_url)
+            list_of_stings = file_path.split("\\")
+            save_log_file(output, list_of_stings[-1].replace('.csv','.txt'))
+            print(output)
+    def process_TSO(self):
+        """
+        Process TSO500 Runs
+        """
+        # destination for the files to be downloaded to:
+        destination_of_Results = config.TSO_destination_of_Results
+        # create folder woth project name
+        for file_path in self.project_files:
+            csv_file = pandas.read_csv(file_path, index_col=None)
+            df_results = csv_file[csv_file['type'] == 'Results']
+            df_coverage = csv_file[csv_file['type'] == 'coverage']
+            df_sompy = csv_file[csv_file['type'] == 'sompy']
+            search_results1 = re.search(r"_(TSO\d+).csv", file_path)
+            search_results2 = re.search(r"002(_\S+)_TSO", file_path)
+            folder_name = (
+            search_results1.group(1) + search_results2.group(1) + "_AUTOMATE_DUTY_TEST"

```

Comment:
this seems different to the other runs?

Reply:
this regex patterns find the patterns in the TSO names in order to generate the TSO file name in the same way as is done manually.